core/src/main/java/oracle/weblogic/deploy/util/WLSBeanHelp.java

```diff
@@ -0,0 +1,565 @@
+      BeanInfo info = getBeanInfo(beanName);
+      if (info == null) return null;
+      for (String pluralize : new String [] { "", "s"}) {
+        String pName = propName + pluralize;
+        for (PropertyDescriptor pd:info.getPropertyDescriptors()) {
+          if (pName.equals(pd.getName())) return pd;
+        }
+        for (PropertyDescriptor pd:info.getPropertyDescriptors()) {
+          if (pName.equalsIgnoreCase(pd.getName())) return pd;
+        }
+      }
+    } catch (Throwable ignore) {}
+    return null;
+  }
+
+  // can return null if not found
+  private static BeanDescriptor getBeanDescriptor(
+    String beanName
+  ) {
+    try {
+      BeanInfo info = getBeanInfo(beanName);
+      if (info == null) return null;
+      return info.getBeanDescriptor();
+    } catch (Throwable ignore) {}
+    return null;
+  }
+
+  private static boolean printAttributeHelp(String beanName, String propName, int margin) {
+    try {
+      BeanInfo info = getBeanInfo(beanName);
+
+      if (info == null)
+        throw new RuntimeException("Error:  Bean '" + beanName + "' not found.");
+
+      for (PropertyDescriptor pd:info.getPropertyDescriptors()) {
+        if (propName.equals(pd.getName())) {
+          println("Bean = " + beanName);
+          println("");
+          printPropertyDescriptor(pd, margin);
+          return true;
+        }
+      }
+
+      println("Error: Prop '" + propName + "' not found in bean '" + beanName + "'.");
+    } catch (Throwable th) {
+      println("Error: " + th.getMessage());
+    }
+    return false;
+  }
+
+  static void print(String s) {
+    System.out.print(s);
+  }
+
+  static void println(String s) {
+    System.out.println(s);
+  }
+
+  static void printPropertyDescriptor(PropertyDescriptor o, int margin) {
+    println("\nPROPERTY\n");
+    println("  name=" + o.getName());
+
+    if (!o.getName().equals(o.getDisplayName()))
+      println("  display name=" + o.getDisplayName());
+
+    if (!o.getName().equals(o.getShortDescription()))
+      println("  short description=" + o.getShortDescription());
+
+    println("  property type=" + o.getPropertyType());
+    println("  hidden=" + o.isHidden());
+
+    for (Enumeration<String> en = o.attributeNames();
+      en.hasMoreElements();) {
+      String s = en.nextElement();
+      Object v = o.getValue(s);
+      if (s.equals("description")) continue;
+      if (s.equals("legalValues")) v = legalValuesAsString(v);
+      if (s.equals("see")) v = legalValuesAsString(v);
+      println("  " + s + "=" + v);
+    }
+
+    println("  " + "description" + "=");
+    println(prettyHTML(o.getValue("description").toString(), margin));
+    println("");
+  }
+
+  public static String prettyHTML(String html, int margin) {
+    return new PrettyHTML(html, margin).toString();
+  }
+
+  public static String prettyHTMLAbbrev(String html, int width) {
+    return new PrettyHTML(html, 1000000).abbrevString(width);
+  }
+
+  public static void prettyHTMLTest() {
```

Comment:
Why we need a test method in the main tree?

Reply:
The content of the unit test requires using a bunch of special tags, so putting it next to the list of tags were testing helps make the unit test self documenting and helps ensure that the unit test covers all tags...

Reply:
Please move this to the proper location in the `src/test` directory tree.  

Also, unit test methods should never need to be public since the class being tested and the unit test class can below to the same package even though they are in different locations.

Reply:
Fixed